Initial device trees for Congo and Morocco platforms.

mahkurap opened the pull request:
ARM64: dts: aspeed: initial device tree for Congo and Morocco platform

This commit chain has the following changes:
- AMD specific flash layout dtsi file
- initial device tree for Congo
- initial device tree for Morocco
- Makefile change.

Tested: Verified on QEMU and Malta/Marley setup

mahkurap committed:
ARM64: dts: aspeed: initial device tree for Congo platform

Adding initial device tree for Congo platform and the corresponding
Makefile change.

Device tree enables the following:
    - eeproms, temp sensors, fans, gpios, and adc, etc.,.

Tested: verified on QEMU with fdt overlay

Signed-off-by: Mahesh Kurapati <mahesh.kurapati@amd.com>

mahkurap committed:
ARM64: dts: aspeed: initial device tree for Morocco platform

Adding initial device tree for Morocco platform and the corresponding
Makefile change.

Device tree enables the following:
    - eeproms, temp sensors, fans, gpios, adc, etc...

Tested: Verified on QEMU with fdt overlay

Signed-off-by: Mahesh Kurapati <mahesh.kurapati@amd.com>

ojayanth commented:
Please update PR description with PR scope .

rajaganeshr commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -0,0 +1,378 @@
+#include <dt-bindings/i2c/i2c.h>
+
+/ {
+	model = "AMD Morocco BMC";
```
can we mention CRB or PRB instead of BMC?

mahkurap replied:
Agreed. Changed to Morocco VRB.

rajaganeshr commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -0,0 +1,378 @@
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+		label = "spi1:0";
```
I think bios-update.sh/fpga-update.sh sometimes check the label string as "pnor". Pls make sure script and dts are aligned.

rajaganeshr commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,289 @@
+#include <dt-bindings/i2c/i2c.h>
+
+/ {
+	model = "AMD Congo BMC";
```
can we rename as PRB/VRB instead of BMC?

mahkurap replied:
changed.

mahkurap committed:
ARM64: dts: aspeed: added amd specific flash layout

Following is the BMC SPI Flash Layout:
 - uboot     0x00      (2MB)
 - uboot-env 0x200000  (128KB)
 - kernel    0x220000  (16MB)
 - rofs      0x1220000 (32MB)
 - rwfs      0x3220000 (79.75MB)

Tested: Verfied on QEMU and Malta/Marley platform

Signed-off-by: Mahesh Kurapati <mahesh.kurapati@amd.com>

ojayanth commented on arch/arm64/boot/dts/aspeed/amd-flash-layout-128.dtsi:
```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-2.0+
```
IUpdate Marley dts with adding new layout?

mahkurap replied:
> IUpdate Marley dts with adding new layout?
ok. will change the Marley dts.

ojayanth commented on arch/arm64/boot/dts/aspeed/Makefile:
```diff
@@ -1,3 +1,4 @@
 dtb-$(CONFIG_ARCH_ASPEED) += \
 	ast2700-evb.dtb \
```
Commit message 
-- Update the device with new updates like gpios etc.
-- Tested should be add qemu validation of of Congo?

ojayanth replied:
Same applicable for morocco commits

mahkurap replied:
ok

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,358 @@
+&uart12 {
+	status = "okay";
+};
+
```
Include sbtsi/sbrmi driver?

mahkurap replied:
they are not needed for initial version for bring-up. Will add them later.

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,358 @@
+                /*Y0-Y7*/       "","","","","","","","",
+                /*Z0-Z7*/       "","","","","","","","",
+                /*AA0-AA7*/     "","","","","","","","",
+                /*AB0-AB7*/     "ASSERT_BMC_READY","P0_MGMT_ASSERT_CLR_CMOS","P0_MGMT_MON_PROCHOT_L","","","","P0_MGMT_ASSERT_NMI_BTN_L","MGMT_SYS_MON_ATX_PWR_OK",
```
Please refer Ashok's PR comments.

mahkurap replied:
ok

mahkurap committed:
ARM64: dts: aspeed: use amd-flash-layout for Marley

This commit has the necessary changes requried in the Marley
device tree to use the amd flash layout. In the amd flash
layout kernel size is increased to 10MB from 9MB, and the rofs
is reduced to 83.5MB from 84.5MB to keep the rwfs at the same
offset.

Tested: Verified on QEMU, boots to Linux

Signed-off-by: Mahesh Kurapati <mahesh.kurapati@amd.com>

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+		status = "okay";
+		m25p,fast-read;
+		label = "pnor";
+		spi-max-frequency = <50000000>;
```
Based on today's call frequency need to updated with verified value. sam e applicable for moroco.

supven01 commented on arch/arm64/boot/dts/aspeed/amd-flash-layout-128.dtsi:
```diff
@@ -0,0 +1,32 @@
+	};
+
+	rofs@b20000 {
+		reg = <0xc20000 0x53E0000>; // 83.875MB
```
Please check if this can be reduced to power of 2 size.

supven01 commented on arch/arm64/boot/dts/aspeed/amd-flash-layout-128.dtsi:
```diff
@@ -0,0 +1,32 @@
+	};
+
+	kernel@220000 {
+		reg = <0x220000 0xA00000>; // 10MB
```
Size = Power of 2 please.

supven01 commented on arch/arm64/boot/dts/aspeed/Makefile:
```diff
@@ -1,3 +1,4 @@
 dtb-$(CONFIG_ARCH_ASPEED) += \
 	ast2700-evb.dtb \
 	aspeed-bmc-amd-marley.dtb \
+	aspeed-bmc-amd-congo.dtb \
```
Alphabetical order please

supven01 commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+#include "amd-flash-layout-128.dtsi"
+	};
+
+	flash@1 {
```
Please check if MP flash is connected to CS1 of BMC FMC controller. if not, remove it.

supven01 commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+		m25p,fast-read;
+		label = "pnor";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
```
Double check please. [single or dual or quad spi lines]

supven01 commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+		label = "pnor";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
```
Double check please. [single or dual or quad spi lines]

supven01 commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+		m25p,fast-read;
+		label = "pnor";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
```
Double check please. [single or dual or quad spi lines]

supven01 commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+		label = "pnor";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
```
Double check please. [single or dual or quad spi lines]

supven01 commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -0,0 +1,371 @@
+                /*Y0-Y7*/       "","","","","","","","",
+                /*Z0-Z7*/       "","","","","","","","",
+                /*AA0-AA7*/     "","","","","","","","",
+                /*AB0-AB7*/     "HPM_FPGA_READY","P0_MGMT_ASSERT_CLR_CMOS",
```
HPM_STBY_EN instead of HPM_FPGA_READY
